TASK-48 Add service controller functionality

KirSanich opened the pull request.

Commit:
TASK-48 Add controller-service support for event lifecycle

Добавление сервисов, контроллеров, мапперов для работы с сущностью события

Commit:
TASK-48 Add controller-service support for event lifecycle

Разрешение merge конлфикта с мастером

Ivanesca commented on core/src/main/kotlin/ru/surf/core/dto/PutRequestEventDto.kt:
```diff
@@ -0,0 +1,14 @@
+    val eventInitiatorId: UUID,
+    val traineesIds: List<UUID?>
+) {
+}
```
То есть чтобы обновить описание нужно передать список со всеми стажерами?

KirSanich replied:
Я пока не знаю, как делать обновление по выбранным полям, с фронта по идее можно передать полную сущность и одни и те же поля просто пересохраняться.

Ivanesca replied:
Можно же просто кастомный запрос в репозитории сделать, и его вместо save вызывать

Ivanesca commented on core/src/main/kotlin/ru/surf/core/controller/EventController.kt:
```diff
@@ -0,0 +1,43 @@
+    fun getEvent(@PathVariable("id") eventId: UUID): ResponseEntity<FullResponseEventDto> {
+        val getResponseEventDto = eventService.getEvent(eventId)
+        return ResponseEntity.ok(getResponseEventDto)
+    }
```
Почему возвращаем только описание?

KirSanich replied:
Это заглушка, ибо конечный вид дто неизвестен, пока фронт не начнем делать

soup1ck replied:
Да, до фронта еще далековато

Ivanesca commented on core/src/main/kotlin/ru/surf/core/controller/EventController.kt:
```diff
@@ -0,0 +1,43 @@
+        val shortResponseEventDto = eventService.updateEvent(eventId, putRequestEventDto)
+        return ResponseEntity.ok(shortResponseEventDto)
+    }
+
```
Тоже только описание?

KirSanich replied:
Заглушка

Ivanesca commented on core/src/main/kotlin/ru/surf/core/controller/EventController.kt:
```diff
@@ -0,0 +1,43 @@
+    fun createEvent(@RequestBody postRequestEventDto: PostRequestEventDto): ResponseEntity<Unit> {
+        eventService.createEvent(postRequestEventDto)
+        return ResponseEntity.ok().build()
+    }
```
Разве при посте мы не должны узнать присвоенный id?

KirSanich replied:
Думал над этим, скорее всего мы отправим id и потом на фронте некст запросом гет, но пока не знаю что на фронте будет.

KirSanich commented on core/src/main/kotlin/ru/surf/core/service/impl/EventServiceImpl.kt:
```diff
@@ -0,0 +1,61 @@
+            traineesAmount = putRequestEventDto.traineesAmount
+            eventType = eventTypeService.getEventType(putRequestEventDto.eventTypeId)
+            eventInitiator = surfEmployeeService.getSurfEmployee(putRequestEventDto.eventInitiatorId)
+            trainees = traineesList
```
Здесь еще нужно засетить какое-то состояние, но я пока не знаю в каком виде с фронта придет, айдишником или какой-то строкой и надо будет маппить.

Commit:
TASK-48 Add controller-service support for event lifecycle(Refactor Put Request)

Рефакторинг дто для обновление ивента, удаление лишней логики связанной со стажерами

Commit:
TASK-48 Add controller-service support for event lifecycle(Fix post/put requests)

Рефакторинг пост и пут запросов
